app/assets/javascripts/discourse/lib/desktop-notifications.js.es6

```diff
@@ -0,0 +1,245 @@
+
+let primaryTab;
+let liveEnabled;
+let notificationTagName;
+let mbClientId;
+let lastAction;
+
+const focusTrackerKey = "focus-tracker";
+const seenDataKey = "seen-notifications";
+const recentUpdateThreshold = 1000 * 60 * 2; // 2 minutes
+const idleThresholdTime = 1000 * 10; // 10 seconds
+const INVITED_TYPE = 8;
+
+function init(container) {
+  liveEnabled = false;
+  requestPermission().then(function() {
+    try {
+      localStorage.getItem(focusTrackerKey);
+    } catch (e) {
+      Em.Logger.info('Discourse desktop notifications are disabled - localStorage denied.');
+      return false;
+    }
+    liveEnabled = true;
+    Em.Logger.info('Discourse desktop notifications are enabled.');
+    return true;
+  }).then(function(c) {
+    if (c) {
+      try {
+        init2(container);
+      } catch (e) {
+        Em.Logger.error(e);
+      }
+    }
+  }).catch(function() {
+    liveEnabled = false;
+    //Em.Logger.debug('Discourse desktop notifications are disabled - permission denied.');
+  });
+}
+
+function init2(container) {
+  // Load up the current state of the notifications
+  const seenData = JSON.parse(localStorage.getItem(seenDataKey));
+  let markAllSeen = true;
+  if (seenData) {
+    const lastUpdatedAt = new Date(seenData.updated_at);
+    if (lastUpdatedAt.getTime() + recentUpdateThreshold > new Date().getTime()) {
+      // The following conditions are met:
+      //  - This is a new Discourse tab
+      //  - The seen notification data was updated in the last 2 minutes
+      // Therefore, there is no need to reset the data.
+      markAllSeen = false;
+    }
+  }
+  if (markAllSeen) {
+    Discourse.ajax("/notifications.json?silent=true").then(function(result) {
+      updateSeenNotificationDatesFrom(result);
+    });
+  }
+
+  notificationTagName = "discourse-notification-popup-" + Discourse.SiteSettings.title;
+
+  const messageBus = container.lookup('message-bus:main');
```

Comment:
If the only thing that the container does is look up the `messageBus`, I think it's cleaner to just pass the `messageBus` instance into the `init` isn't it? Why give this code the ability to look up anything when it just needs one thing?